python_modules/libraries/dagster-dbt/dagster_dbt/core/resources_v2.py

```diff
@@ -887,82 +894,55 @@ def _fetch_and_attach_row_count_metadata(
+            # Avoid counting rows for views, since they may include complex SQL queries
+            # that are costly to execute. We can revisit this in the future if there is
+            # a demand for it.
+            if is_view:
+                return event
 
-            return event.with_metadata(metadata={**event.metadata, **additional_metadata})
-        except Exception as e:
-            logger.exception(
-                f"An error occurred while fetching row count for {unique_id}. Row count metadata"
-                " will not be included in the event.\n\n"
-                f"Exception: {e}"
-            )
-            return event
+            unique_id = dbt_resource_props["unique_id"]
+            logger.debug("Fetching row count for %s", unique_id)
+            table_str = f"{dbt_resource_props['database']}.{dbt_resource_props['schema']}.{dbt_resource_props['name']}"
 
-    def _stream_dbt_events_and_enqueue_postprocessing(
-        self,
-        output_events_and_futures: List[Union[Future, DbtDagsterEventType]],
-        executor: ThreadPoolExecutor,
-    ) -> None:
-        """Task which streams dbt events and either directly places them in
-        the output_events list to be emitted to user code, or enqueues post-processing tasks
-        where needed.
-        """
-        for event in self:
-            # For any materialization or output event, we run postprocessing steps
-            # to attach additional metadata to the event.
-            if isinstance(event, (AssetMaterialization, Output)):
-                output_events_and_futures.append(
-                    executor.submit(
-                        self._fetch_and_attach_row_count_metadata,
-                        event,
+            try:
+                with adapter.connection_named(f"row_count_{unique_id}"):
+                    query_result = adapter.execute(
+                        f"""
+                            SELECT
+                            count(*) as row_count
+                            FROM
+                            {table_str}
+                        """,
+                        fetch=True,
                     )
+                row_count = query_result[1][0]["row_count"]
+
+                new_metadata = {
+                    **TableMetadataSet(row_count=row_count),
+                }
+                return event.with_metadata({**event.metadata, **new_metadata})
+
+            except Exception as e:
+                logger.exception(
+                    f"An error occurred while fetching row count for {unique_id}. Row count metadata"
+                    " will not be included in the event.\n\n"
+                    f"Exception: {e}"
                 )
-            else:
-                output_events_and_futures.append(event)
+                return event
 
     @public
     @experimental
     def fetch_row_counts(
-        self,
+        self, *, num_threads=DEFAULT_EVENT_POSTPROCESSING_THREADPOOL_SIZE
```

Comment:
This feels off, since this implies that `num_threads` would also be an argument for other methods that emit metadata for the `DbtEventIterator` (e.g. `.fetch_column_schema`).

Could we instead just have it as a property on `DbtCliInvocation`?

Reply:
I was planning to have it as a kwarg param on each of the chained/builder methods, so users could specify a different threadpool size for each.

The previous shared threadpool approach is tricky because of its lifecycle - there's no clear "owner" or `with`-context scope we can use. Could move the configuration option to `DbtCliInvocation` but still have separate threadpools?

Reply:
> Could move the configuration option to DbtCliInvocation but still have separate threadpools?

Yeah I think this is preferable.